lib/elixir/lib/exception.ex

```diff
@@ -788,6 +788,10 @@ defmodule BadFunctionError do
   defexception [:term]
 
   @impl true
+  def message(%{term: term}) when is_function(term) do
+    "function #{term} does not exist in this application version"
```

Comment:
We need a test for this because it is missing `#{inspect(term)}`. Here is a minimal script to reproduce it:

```elixir
defmodule Foo do
  def fun, do: fn -> :ok end
end

fun = Foo.fun

defmodule Foo do
  def fun, do: fn -> :another end
end

:code.purge(Foo)
fun.()
```

Also, this is not related to the application version. It just happens it can be triggered by application versions. My suggestion would be something similar to what you proposed:

> function #{inspect(term)} is invalid, likely because it points an old version of the code